### PR TITLE
modified cocitation and bibcouple recommenders for luigi

### PR DESCRIPTION
**Pull request by rohanaras**

### CODE REVIEW

File: babel_util/recommenders/bibcouple.py [hunk 1]
```diff
@@ -2,34 +2,30 @@
...
     import numpy as np
-    parser = argparse.ArgumentParser()
-    parser.add_argument('infile', nargs='?', type=argparse.FileType('r'), default=sys.stdin)
-    parser.add_argument('outfile', nargs='?', type=argparse.FileType('w'), default=sys.stdout)
-    parser.add_argument('dimension', type=int, help="Dimension of matrix. Matrix is square")
-    parser.add_argument('-n', type=int, help="Max number of recommendations to generate per-paper", default=10)
-    parser.add_argument('-d', '--delimiter', type=str, help="Delimiter used in the link file", choices=FILE_DELIM, default=',')
 
-    args = parser.parse_args()
+    if not cmdline:
```
Comment (iwsmith) [inline babel_util/recommenders/bibcouple.py]: Make it clear what cmdline does.

File: babel_util/recommenders/bibcouple.py [hunk 2]
```diff
@@ -39,13 +35,40 @@ def get_next_id():
...
-            args.outfile.write("{0} {1} {2}\n".format(paper_ids[i],
-                                                      paper_ids[entry[0]],
-                                                      entry[1]))
+        for entry in recs[:numRecs]:
+            sEntry = "{0} {1} {2}\n".format(paper_ids[i], paper_ids[entry[0]], entry[1])
+            if cmdline:
+                outfile.write(sEntry)
+            else:
+                returnString.append(sEntry)
+    if not cmdline:
```
Comment (iwsmith) [inline babel_util/recommenders/bibcouple.py]: If you have to build an in memory string buffer use StringIO, not a list of strings. Also figure out how Luigi will let us write to file, some of these files are quite large and will use up all memory.

File: babel_util/recommenders/bibcouple.py [hunk 3]
```diff
@@ -39,13 +35,40 @@ def get_next_id():
...
+        return ''.join(returnString)
+
+def peek_line(f):
```
Comment (iwsmith) [inline babel_util/recommenders/bibcouple.py]: Pull this out

**Comment by rohanaras** [comment]
Everything should work now

File: babel_util/recommenders/bibcouple.py [hunk 4]
```diff
@@ -45,7 +38,20 @@ def get_next_id():
         # A paper shouldn't recommend itself
         recs = filter(lambda x: x[0] != i, recs)
         recs.sort(key=lambda x: x[1], reverse=True)
-        for entry in recs[:args.n]:
-            args.outfile.write("{0} {1} {2}\n".format(paper_ids[i],
-                                                      paper_ids[entry[0]],
-                                                      entry[1]))
+        for entry in recs[:numRecs]:
+            sEntry = "{0} {1} {2}\n".format(paper_ids[i], paper_ids[entry[0]], entry[1])
```
Comment (iwsmith) [inline babel_util/recommenders/bibcouple.py]: This variable in unneeded, remove please.

**Pull request merged**